catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogCapability.java

```diff
@@ -31,4 +31,19 @@ public CapabilityResult managedStorage(Scope scope) {
     return CapabilityResult.unsupported(
         String.format("Hadoop catalog does not support managed storage for %s.", scope));
   }
+
+  @Override
+  public CapabilityResult specificationOnName(Scope scope, String name) {
+    CapabilityResult capabilityResult = Capability.super.specificationOnName(scope, name);
+    if (!capabilityResult.supported()) {
+      return capabilityResult;
+    }
+
+    if (name.contains("/")) {
```

Comment:
Thanks for adding this check.
I'm thinking if we could promote things like this to project-wide naming constraints.
In Kubernetes, for example, we always check if a given name is a valid DNS subdomain name, sometimes we even require a DNS label. The rationale behind this is that we don't want unexpected characters in names that appear in URIs. They are simply too dangerous. And ... there is always a bound for name lengths.

Reply:
What you described is exactly the responsibility of interface `Capability`:
https://github.com/apache/gravitino/blob/9a5fef924ee3488ef99bb00c75913f885a84dff3/core/src/main/java/org/apache/gravitino/connector/capability/Capability.java#L74-L83

which will be implemented by each catalog. It already has a default implementation with constraints on length and special characters. I think it is considered strict enough.